Work-around for latexml bug for MyLicense.txt

HansOlsson opened the pull request.

henrikt-ma reviewed (changes requested):
Providing a better fix with no impact on PDF generation, and minimal – if any – impact on HTML generation.

henrikt-ma commented:
I tried finding an issue for this in the LaTeXML tracker, but couldn't find one.  Has anyone else been able to find the issue, or should I report it?

HansOlsson commented:
I have now updated to use ifpdf, but with comment separated and link to modelica-issue, so that we don't have to update it when reported and fixed for Latexml.

henrikt-ma commented on chapters/annotations.tex:
```diff
@@ -2538,7 +2538,15 @@ \subsection{Licensing}\label{licensing}
 \section{License Texts}\label{license-texts}
 
-For a top-level class the {\lstinline!annotation(License="modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!}\annotationindex{License}, gives a license text file for the class.
+% TODO: The following works around HTML-issue with spurious percent-sign in .txt https://github.com/modelica/ModelicaSpecification/issues/3663
```
I opened a proper LaTeXML issue which is what we should refer to here:
```suggestion
% Need to work around LaTeXML issue with long \lstinline argument:
% - https://github.com/brucemiller/LaTeXML/issues/2553
```

henrikt-ma commented on chapters/annotations.tex:
```diff
@@ -2538,7 +2538,15 @@ \subsection{Licensing}\label{licensing}
+\lstinline!annotation(License="modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!%
+\else
+\lstinline!annotation(License="$\text{modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt}$")!%
+\fi\annotationindex{License}, gives a license text file for the class.
```
Better to not have content on the same line as `\fi`:
```suggestion
\fi%
\annotationindex{License}, gives a license text file for the class.
```

henrikt-ma reviewed (changes requested):
Now that it is understood that the problem has to do with the length of, maybe there is a better workaround…

HansOlsson commented:
> Now that it is understood that the problem has to do with the length of, maybe there is a better workaround…

See #3685

henrikt-ma commented on chapters/annotations.tex:
```diff
@@ -2538,7 +2538,17 @@ \subsection{Licensing}\label{licensing}
 
-For a top-level class the {\lstinline!annotation(License="modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!}\annotationindex{License}, gives a license text file for the class.
+% Need to work around LaTeXML issue with long \lstinline argument:
+% - https://github.com/brucemiller/LaTeXML/issues/2553
+% Clean up when corrected.
+%
+For a top-level class the
+\ifpdf
+\lstinline!annotation(License="modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!%
+\else
+\lstinline!annotation(License="$\text{modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt}$")!%
```
Fix which is more to the point:
```suggestion
For a top-level class the
% Workaround for LaTeXML issue with long arguments to \lstinline:
% - https://github.com/brucemiller/LaTeXML/issues/2553
\ifpdf
\lstinline!annotation(License="modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!%
\else
\lstinline!annotation(License=!%
\lstinline!"modelica:/ModelicaLibraryName/Resources/Licenses/MyLicense.txt")!%
```

HansOlsson commented:
Replaced by #3685
